src/main/java/com/google/devtools/build/lib/analysis/test/TestActionBuilder.java

```diff
@@ -366,8 +364,7 @@ private TestParams createTestAction()
     ImmutableList.Builder<Artifact> coverageArtifacts = ImmutableList.builder();
     ImmutableList.Builder<ActionInput> testOutputs = ImmutableList.builder();
 
-    ActionOwner actionOwner =
-        testConfiguration.useTargetPlatformForTests() ? getTestActionOwner() : getOwner();
+    ActionOwner actionOwner = getTestActionOwner(testConfiguration.useTargetPlatformForTests());
```

Comment:
This was an unrelated refactoring I did while tracing @AustinSchuh's past work.
It's a no-op change, but it helped me trace the current logic related to execution requirements a bit easier.

I can drop the hunks if folks don't like it.

Reply:
I like pushing the logic down into `getTestActionOwner`.